src/core/Form/DateInput/DateInput.tsx

```diff
@@ -0,0 +1,482 @@
+
+  return (
+    <HtmlDiv
+      {...wrapperProps}
+      className={classnames(baseClassName, className, {
+        [dateInputClassNames.disabled]: !!passProps.disabled,
+        [dateInputClassNames.error]: status === 'error',
+        [dateInputClassNames.success]: status === 'success',
+        [dateInputClassNames.fullWidth]: fullWidth,
+        [dateInputClassNames.hasPicker]: datePickerEnabled,
+      })}
+    >
+      <HtmlSpan className={dateInputClassNames.styleWrapper}>
+        <Label
+          htmlFor={id}
+          labelMode={labelMode}
+          optionalText={optionalText}
+          className={classnames({
+            [dateInputClassNames.labelIsVisible]: labelMode !== 'hidden',
+          })}
+          tooltipComponent={tooltipComponent}
+        >
+          {labelText}
+        </Label>
+        <HintText id={hintTextId}>{hintText}</HintText>
+        <HtmlDiv className={dateInputClassNames.inputAndPickerWrapper}>
+          <HtmlDiv className={dateInputClassNames.inputElementContainer}>
+            <Debounce waitFor={debounce}>
+              {(debouncer: Function) => (
+                <HtmlInput
+                  autoComplete="off"
+                  {...passProps}
+                  value={'value' in props ? value : inputValue}
+                  id={id}
+                  className={dateInputClassNames.inputElement}
+                  forwardedRef={inputRef}
+                  placeholder={visualPlaceholder}
+                  aria-invalid={status === 'error' ? true : undefined}
+                  {...getConditionalAriaProp('aria-describedby', [
+                    statusText ? statusTextId : undefined,
+                    hintText ? hintTextId : undefined,
+                    ariaDescribedBy,
+                  ])}
+                  onChange={(event: ChangeEvent<HTMLInputElement>) => {
+                    const eventValue = event.currentTarget.value;
+                    conditionalSetInputValue(eventValue);
+                    if (propOnChange) {
+                      debouncer(propOnChange, {
+                        value: eventValue,
+                        date: dateAdapter.parse(eventValue),
+                      });
+                    }
+                  }}
+                />
+              )}
+            </Debounce>
+          </HtmlDiv>
+          {datePickerEnabled && (
+            <HtmlDiv className={dateInputClassNames.pickerElementContainer}>
+              <HtmlButton
+                forwardedRef={openButtonRef}
+                className={classnames(dateInputClassNames.pickerButton, {
+                  [dateInputClassNames.pickerButtonDisabled]:
+                    passProps.disabled,
+                })}
+                onClick={() => toggleCalendar(!calendarVisible)}
+                disabled={passProps.disabled}
+              >
+                <VisuallyHidden>
+                  {texts.openButtonLabel}
+                  {buttonDateLabel}
+                </VisuallyHidden>
```

Comment:
Same thing as line 254. Makes screen reader (at least VoiceOver) take a small break between words
```suggestion
                <VisuallyHidden>
                  {texts.openButtonLabel} - {buttonDateLabel}
                </VisuallyHidden>
```

Reply:
Good point! This [ARIA practices datepicker example](https://w3c.github.io/aria-practices/examples/dialog-modal/datepicker-dialog.html) uses comma as a separator, which sounded good when testing with NVDA

Reply:
Updated with comma